packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -116,10 +156,116 @@ export function CrossmintWalletProvider({
         return { startedCreation: true };
     };
 
+    const createPasskeyPrompt = (type: ValidPasskeyPromptType) => () =>
+        new Promise<void>((resolve) => {
+            if (!showPasskeyHelpers) {
+                resolve();
+                return;
+            }
+            setPasskeyPromptState({
+                type,
+                open: true,
+                primaryActionOnClick: () => {
+                    setPasskeyPromptState({ open: false });
+                    resolve();
+                },
+                secondaryActionOnClick: () => {
+                    setPasskeyPromptState({ open: false });
+                    resolve();
+                },
+            });
+        });
+
     const clearWallet = () => {
         setWalletState({ status: "not-loaded" });
     };
 
+    const getPasskeySigner = async () => {
+        const cache = getCache();
+        if (cache === null || cache.passkey === undefined) {
+            // Create a new passkey
+            const passkeyName = "Crossmint Wallet";
+            const passkeySigner = await createPasskeySigner(passkeyName);
+            updateCache({
+                passkey: {
+                    authenticatorId: passkeySigner.credential.id,
+                    domain: "localhost",
+                    passkeyName: passkeyName,
+                    pubKeyPrefix: passkeySigner.credential.publicKey.prefix,
+                    pubKeyX: passkeySigner.credential.publicKey.x.toString(),
+                    pubKeyY: passkeySigner.credential.publicKey.y.toString(),
+                },
+            });
+            return passkeySigner;
+        }
+        return {
+            type: "PASSKEY",
+            credential: {
+                id: cache.passkey.authenticatorId,
```

Comment:
if you just created a passkey, wouldn't `cache.passkey.authenticatorId` be null as `cache` for fetched before calling `updateCache`?

Reply:
Assuming you mean the `if (cache === null || cache.passkey === undefined) {` condition just above that, there's an early return that outputs the newly created passkey.

In other words, if we've just created a passkey, this code won't be reached.

Reply:
ah missed that!